jbi/queue.py

```diff
@@ -0,0 +1,207 @@
+import itertools
+import json
+import logging
+import os
+import shutil
+import traceback
+from abc import ABC
+from collections import defaultdict
+from datetime import datetime, timezone
+from functools import lru_cache
+from pathlib import Path
+from typing import Optional
+from urllib.parse import urlparse
+
+from pydantic import BaseModel, Field
+
+from jbi import bugzilla
+from jbi.environment import get_settings
+
+logger = logging.getLogger(__name__)
+
+
+class PythonException(BaseModel, frozen=True):
+    type: str
+    description: str
+    details: str
+
+    @classmethod
+    def from_exc(cls, exc: Exception):
+        return PythonException(
+            type=exc.__class__.__name__,
+            description=str(exc),
```

Comment:
For easier debugging I think we could dump the response status and body for `HTTPError`, with a `isinstance(exc, HTTPError) and hasattr(exc, "response")`